### PR TITLE
Condense mtarget UI, add editor

### LINKED ISSUE #1726: Add ability to add comments to each DMG instance in a multi opt
This would better allow people to describe assumptions for each hit of a multi opt, such as which buffs are applying

### PR DESCRIPTION
**Pull request by frzyc**
## Describe your changes

Condense mtarget UI
- Add a floating target editor
- Reduce the amount of UI element per target
- Convert some of the inputs to TextFields

## Issue or discord link

- Resolves #1726

## Testing/validation

image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Review by nguyentvan7 — DISMISSED** [review]
image
With a lot of bonus stats, the editor can take up too much space. bit of an edge case, but I could see that being an issue for more complex multi-opt

**Comment by frzyc** [comment]
> image With a lot of bonus stats, the editor can take up too much space. bit of an edge case, but I could see that being an issue for more complex multi-opt

Oh yes, good point. I can make it so that clicking on "Target Editor" collapse the editor

File: libs/gi/page-team/src/CharacterDisplay/CustomMultiTarget/MTargetEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            flexGrow: 1,
+            gap: 1,
+            height: '100%',
+            py: 1,
+            alignItems: 'center',
+          }}
+          onClick={() => setcollapse((c) => !c)}
+        >
+          <Typography variant="h6">Target Editor</Typography>
+          {collapse ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
```
Comment (nguyentvan7) [inline libs/gi/page-team/src/CharacterDisplay/CustomMultiTarget/MTargetEditor.tsx:146]: Is there a way to expand autoamtically when you select a target? Seems like a logical move

**Review by nguyentvan7 — APPROVED** [review]

File: libs/gi/page-team/src/CharacterDisplay/CustomMultiTarget/MTargetEditor.tsx (hunk 1 above)
Comment (frzyc) [reply libs/gi/page-team/src/CharacterDisplay/CustomMultiTarget/MTargetEditor.tsx:146]: That makes sense.

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**